src/main/java/edu/nitt/delta/bustracker/config/WebSecurityConfig.java

```diff
@@ -1,15 +1,76 @@
 package edu.nitt.delta.bustracker.config;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import edu.nitt.delta.bustracker.service.BusTrackerUserDetailsService;
+import edu.nitt.delta.bustracker.utils.JwtTokenUtil;
 
 @Configuration
 @EnableWebSecurity
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private BusTrackerUserDetailsService busTrackerUserDetailsService;
+
+    @Autowired
+    private PasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    private BusTrackerFilter busTrackerFilter;
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.csrf().disable().authorizeRequests().anyRequest().permitAll();
+        http.csrf()
+            .disable()
+            .authorizeRequests()
+            .antMatchers("/login")
+            .permitAll()
+            .anyRequest()
+            .authenticated();
```

Comment:
Since you've configured the built-in authority/role-based auth, use it for the admin routes with `antMatchers(<admin routes>).hasRole(Roles.ADMIN.toString())`